lib/testsuite/index.js

```diff
@@ -668,11 +668,11 @@ class TestSuite {
         httpOutput: Logger.collectOutput()
       }));
     } else if (process.port && typeof process.port.postMessage === 'function') {
-      process.port.postMessage({
+      process.port.postMessage(SafeJSON.stringify({
         type: 'testsuite_finished',
         results: this.reporter.exportResults(),
         httpOutput: Logger.collectOutput()
-      });
+      }));
```

Comment:
We can refactor `sendReportToParentWorker`. In both child process and worker threads we are dealing with same object  so perhaps we can only call `SafeJson.stringify once.`

Reply:
There is one extra field in the child process block (`itemKey`), so can't stringify them together.